code/modules/cm_marines/Xeno_Research/Xeno_Analyzation.dm

```diff
@@ -0,0 +1,257 @@
+//// Research Stuff ////
+/obj/machinery/alien_container
+	name = "Speciemen Analyze Chamber"
+	desc = "Standart pressure chamber for all your biology needs"
+	icon = 'icons/Marine/Research/Research_Machinery.dmi'
+	icon_state = "analyzer_chamber"
+	var/mob/living/carbon/Xenomorph/occupant = null
+	var/obj/machinery/computer/analyze_console/linked_console = null
+
+	use_power = 1
+	idle_power_usage = 20
+	active_power_usage = 200
+
+/obj/machinery/alien_container/update_icon()
+	if(occupant)
+		icon_state = "analyzer_chamber_full"
+		return
+	icon_state = "analyzer_chamber"
+
+/obj/machinery/alien_container/attackby(obj/item/W, mob/living/user)
+	if(istype(W, /obj/item/grab))
+		var/obj/item/grab/G = W
+		if(!ismob(G.grabbed_thing))
+			return
+		var/mob/M = G.grabbed_thing
+		if(isXeno(M))
+			put_mob(M)
+	return
+
+/obj/machinery/alien_container/proc/put_mob(mob/living/carbon/M as mob)
+	if (stat & (NOPOWER|BROKEN))
+		to_chat(usr, "\red The chamber is not functioning.")
+		return
+	if (!istype(M))
+		to_chat(usr, "\red <B>The chamber cannot handle such a lifeform!</B>")
+		return
+	if (!isXeno(M))
+		to_chat(usr, "\red <B>You feel stupid...</B>")
+		return
+	if (occupant)
+		to_chat(usr, "\red <B>The chamber is already occupied!</B>")
+		return
+	if (M.abiotic())
+		to_chat(usr, "\red Subject may not have abiotic items on.")
+		return
+	M.forceMove(src)
+	occupant = M
+	update_use_power(2)
+//	M.metabslow = 1
+	add_fingerprint(usr)
+	update_icon()
+	return 1
+
+/obj/machinery/alien_container/verb/move_eject()
+	set name = "Eject occupant"
+	set category = "Object"
+	set src in oview(1)
+	if(usr == occupant)//If the user is inside the tube...
+		to_chat(usr, "\blue Release sequence activated. This will take two minutes.")
+		sleep(1200)
```

Comment:
Use `do_after()` unless the machine isn't anchored and can be moved around, thanks.